### ISSUE TITLE
About the uspto50k P2R model checkpoint

### ISSUE THREAD

**Issue by fiberleif** [otori-bird/retrosynthesis#11]
Dear authors, 

Thanks for sharing the code, and some checkpoints about this great work. l am wondering how you trained these checkpoint (i.e., `models/P2R/USPTO_50K_P2R.pt`), via 1) pretrain-finetune, or 2) train from scratch?

![image](https://github.com/otori-bird/retrosynthesis/assets/11922971/b813fbb7-8bd8-4b41-92b7-6e64c0d9716b)

**Comment by otori-bird** [comment]
This checkpoint is obtained by pretrain-fintune formula.

However, if obtained by train from scratch, you should be able to get a close prediction result.

**Comment by fiberleif** [comment]
Thank you! may l ask which pretrained checkpoint you were using? since I observed that the file size of the shared pretrained checkpoints is ~500MB, but the fine-tuned checkpoint here is only ~200MB, which is not consistent with the checkpoint size

**Comment by otori-bird** [comment]
> Thank you! may l ask which pretrained checkpoint you were using? since I observed that the file size of the shared pretrained checkpoints is ~500MB, but the fine-tuned checkpoint here is only ~200MB, which is not consistent with the checkpoint size.
> 
> ![image](https://private-user-images.githubusercontent.com/11922971/289722948-c73ae238-82f6-49ad-a113-d9cec10924bb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIzNjE4MjUsIm5iZiI6MTcwMjM2MTUyNSwicGF0aCI6Ii8xMTkyMjk3MS8yODk3MjI5NDgtYzczYWUyMzgtODJmNi00OWFkLWExMTMtZDljZWMxMDkyNGJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjEyVDA2MTIwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3MDhiZDJlYjdiYzc5NDZiZjBmOTcwZGUzMjhjY2QxZGZiMjUyYmEwOGQ2YjI0MzBiYjc0MDE5NWJkY2IyMjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ByvlIJ8WXHNtdfZSGq2jF5NQiuXNVNGQsLTxlidl09c)

I apologize for the confusion. 
The checkpoints in the "pretrained" folder are all pre-trained, while the remaining checkpoints are trained from scratch.

If you are looking to obtain fine-tuned checkpoints, it may require you to perform the fine-tuning by yourself. 
Upon checking my local repository and disk storage, it appears that the fine-tuned checkpoints have been inadvertently lost or misplaced.
I apologize for any inconvenience this may have caused you.

**Comment by fiberleif** [comment]
Thank you for your detailed reply, and now l know where the checkpoints come from exactly. May I ask one more question about the role of pretraining in R-SMILES? If the models by "training from scratch" are very close to that by "pretrain-fintune formula", why we need pretraining?

**Comment by otori-bird** [comment]
> Thank you for your detailed reply; now I understand how the checkpoints are trained in detail. May I ask one more question regarding the role of pretraining in R-SMILES? If the models trained from scratch are very close to those trained using the pretrain-finetune formula, what is the point of pretrainig models here?
> 
> Thanks

This is a good question. It can be attributed to two main points: 
(1) It does indeed bring about certain performance improvements. In our local experimental environment, the model trained from scratch achieved top-1 and top-10 accuracy rates of 55.0% and 89.8% respectively, while the model fine-tuned from pretraining achieved top-1 and top-10 accuracy rates of 56.3% and 91.0% respectively; 
(2) It significantly reduces the training time on large datasets. Training a model from scratch on large datasets like uspto-full is very time-consuming, whereas the training time for fine-tuning based on a pretrained model is much less.

**Comment by otori-bird** [comment]
Since the questioner has not asked any further questions for a long time, I believe it is appropriate to close this issue.
